src/app/modules/item/components/thread/thread.component.ts

```diff
@@ -38,21 +57,41 @@ export class ThreadComponent implements OnChanges, OnDestroy {
     private groupWatchingService: GroupWatchingService,
   ) {}
 
-  ngOnChanges(): void {
-    if (!this.itemData) return;
+  ngOnChanges(changes: SimpleChanges): void {
+    if (!this.itemData || !changes.itemData) return;
+    const previousValue = changes.itemData.previousValue as ItemData | undefined;
+    if (previousValue && this.itemData.route.id !== previousValue.route.id) this.threadService.unfollow();
     this.itemData$.next(this.itemData);
```

Comment:
1. This line assumes there is only `itemData` as input. That is true now but this may change, so it would be safe to check.

2. Then, don't forget that `itemData` builds up, so it will change while you do not want to change anything to the thread service, so maybe having only the `itemId` as input would make more sense?

3. Also, in order to convince oneself that the follow and unfollow are called as expected, they should be called exactly at the same / from the same trigger. Here `unfollow` is called in `ngOnChanges` while `follow` is called in the `this.itemData$` subscription.

Reply:
I changed to itemId :+1: 
For the point 3, I used the `pairwise` operator to avoid calling `unfollow` on the first visited item.